contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpGroupScan.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http;
+
+import java.util.List;
+import org.apache.drill.common.expression.SchemaPath;
+
+import org.apache.drill.exec.physical.base.AbstractGroupScan;
+import org.apache.drill.exec.physical.base.GroupScan;
+import org.apache.drill.exec.physical.base.PhysicalOperator;
+import org.apache.drill.exec.physical.base.ScanStats;
+import org.apache.drill.exec.physical.base.ScanStats.GroupScanProperty;
+import org.apache.drill.exec.physical.base.SubScan;
+import org.apache.drill.exec.proto.CoordinationProtos.DrillbitEndpoint;
+import org.apache.drill.shaded.guava.com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpGroupScan extends AbstractGroupScan {
+  private static final Logger logger = LoggerFactory.getLogger(HttpGroupScan.class);
+
+  private final List<SchemaPath> columns;
+  private final HttpScanSpec httpScanSpec;
+  private final HttpStoragePluginConfig httpStoragePluginConfig;
+  private boolean filterPushedDown = true;
+
+  public HttpGroupScan (
+    HttpStoragePluginConfig config,
+    HttpScanSpec scanSpec,
+    List<SchemaPath> columns
+  ) {
+    super("no-user");
+    this.httpStoragePluginConfig = config;
+    this.httpScanSpec = scanSpec;
+    this.columns = columns == null || columns.size() == 0 ? ALL_COLUMNS : columns;
+  }
+
+  public HttpGroupScan(HttpGroupScan that) {
+    super(that);
+    httpStoragePluginConfig = that.getStorageConfig();
+    httpScanSpec = that.getScanSpec();
+    columns = that.getColumns();
+  }
+
+  @Override
+  public void applyAssignments(List<DrillbitEndpoint> endpoints) {
+    logger.debug("HttpGroupScan applyAssignments");
+  }
+
+  @Override
+  public int getMaxParallelizationWidth() {
+    return 1;
+  }
+
+
+  @Override
+  public boolean canPushdownProjects(List<SchemaPath> columns) {
+    return true;
+  }
+
+  @Override
+  public SubScan getSpecificScan(int minorFragmentId) {
+    logger.debug("HttpGroupScan getSpecificScan");
+    return new HttpSubScan(httpStoragePluginConfig, httpScanSpec, columns);
+  }
+
+  @Override
+  public GroupScan clone(List<SchemaPath> columns) {
+    logger.debug("HttpGroupScan clone {}", columns);
+    return new HttpGroupScan(this);
+  }
+
+  @Override
+  public String getDigest() {
+    return toString();
+  }
+
+  @Override
+  public List<SchemaPath> getColumns() {
+    return columns;
+  }
+
+  @Override
+  public PhysicalOperator getNewWithChildren(List<PhysicalOperator> children) {
+    Preconditions.checkArgument(children.isEmpty());
+    return new HttpGroupScan(this);
+  }
+
+  @Override
+  public ScanStats getScanStats() {
```

Comment:
I have a question here.  Can you suggest a good way to guess the estimated row count?  Some APIs are likely to produce very small ones and others not so much. My goal with this plugin was for it really to be used with smaller APIs but there's no reason this couldn't be used with larger ones.  Should I just pick an arbitrary number?  My sense is that this falls in the category of you can't predict the future.

Reply:
I removed all the filter pushdown code and added some rudimentary code to calculate the scanStats.

Reply:
What if we put a boolean config option `expectLargeResults` or something like that, and in the docs, simply state that this is to help Drill optimize these queries.  If the API is likely to return results of `n` or greater, set to `true` otherwise, set to `false`?  Would that make sense?   

If that's the case, would 10k be the boundary to broadcast or not?

Reply:
For now, I updated the PR with a guesstimate of 10K rows. A REST call is unlikely to return more than that in one request. Let's see how well this works in practice.